verify

```diff
@@ -125,6 +125,13 @@ function verify_rpm() {
 	# install all non-source packages
 	(
 		set -x
+		product_version=$(source /etc/os-release; echo "$REDHAT_SUPPORT_PRODUCT_VERSION")
+		if [ "$product_version" = "rawhide" ]; then
+				# force $releasever to account for Fedora pre-release images, as they
+				# may still be using "rawhide", which is not present on our package
+				# repositories on download.docker.com.
+				export DNF_VAR_releasever=${DIST_VERSION}
+		fi
 		${pkg_manager} install -y ${packages}
 	)
```

Comment:
Since we're already in a subshell, here's what I'd suggest (also accounting for `set -u`, which we don't currently set in this script that I can see, but might want to in the future):

```suggestion
	(
		source /etc/os-release # only dangerous if you think it might define `DIST_VERSION` in the future 😅
		set -x
		if [ "${REDHAT_SUPPORT_PRODUCT_VERSION:-}" = 'rawhide' ]; then
				# force $releasever to account for Fedora pre-release images, as they
				# may still be using "rawhide", which is not present on our package
				# repositories on download.docker.com.
				export DNF_VAR_releasever="$DIST_VERSION"
		fi
		${pkg_manager} install -y ${packages}
	)
```

Alternatively:

```suggestion
	(
		set -x
		product_version=$(source /etc/os-release; echo "${REDHAT_SUPPORT_PRODUCT_VERSION:-}")
		if [ "$product_version" = 'rawhide' ]; then
				# force $releasever to account for Fedora pre-release images, as they
				# may still be using "rawhide", which is not present on our package
				# repositories on download.docker.com.
				export DNF_VAR_releasever="$DIST_VERSION"
		fi
		${pkg_manager} install -y ${packages}
	)
```

Reply:
> only dangerous if you think it might define `DIST_VERSION` in the future 😅

Well, that was roughly my concern (it potentially overwriting existing env-vars, which could be a "hard to locate" issue) 😅